proposals/collection-expression-arguments.md

```diff
@@ -268,3 +268,44 @@ strongly triggers the view that this is simply an implicit-object-creation.  And
 case where a constructor *is* actually called (like for `Dictionary<,>`) it is misleading when calling a *create
 method*, or creating an interface.  Finally, there is general apprehension around using `new` at all as there
 is a feeling of redundancy around both the `new` indicating a new instance, *and* `[...]` indicating a new instance.
+
+## Conversions
+
+Collection arguments are *not* considered when determining *collection expression* conversions.
+
+## Construction
+
+Construction is updated as follows.
+
+The elements of a collection expression are evaluated in order, left to right.
+Within *collection arguments*, the arguments are evaluated in order, left to right.
+Each element or argument is evaluated exactly once, and any further references refer to the results of this initial evaluation.
+
+If *collection_arguments* is not the first element in the collection expression, an error is reported.
+
+If the target type is an *array*, *span*, *array interface*, or *generic parameter type*, and the *argument list* is not empty, a binding error is reported.
```

Comment:
i would be ok saying that it's an error to have the element, even if had an empty set of arguments.  these guys do not take args at all, so providing them (even 0 length) seems confusing :)

Reply:
The array interface `IList<T>` causes us to construct `List<T>`, and we should allow `args(capacity: ...)` then. You'd mentioned wanting `args(comparer: ...)` for the IReadOnlyDictionary interface.

Reply:
I think I prefer only one of these alternatives:

- always disallow `with()` (i.e. require at least one arg)
- always allow `with()`

I don't like that allowing `with()` would fluctuate based on target type or based on how the compiler evolves to permit new args to existing target types in the future.